base/setup/lib/setuplib.c

```diff
@@ -1033,6 +1029,16 @@ InitializeSetup(
         DPRINT1("SourceRootPath (2): '%wZ'\n", &pSetupData->SourceRootPath);
         DPRINT1("SourceRootDir (2): '%wZ'\n", &pSetupData->SourceRootDir);
 
+        /* Retrieve the target machine architecture type */
+        // FIXME: This should be determined at runtime!!
+        // FIXME: Allow for (pre-)installing on an architecture
+        //        different from the current one?
+#if defined(SARCH_PC98)
+        pSetupData->ArchType = ARCH_NEC98x86;
```

Comment:
Please use the `IsNEC_98` macro, which is designed for this purpose.

Reply:
Addressed